hooks/configure

```diff
@@ -14,120 +14,147 @@
+    """Return true if core_support interface is available
+
+    The check is done indirectly via checking if systemctl can be used.
+    """
+    return systemctl("--version").returncode == 0
+
+
+def switch_handle_power_key(action):
+    valid_actions = ("ignore","poweroff","reboot","halt","kexec","suspend","hibernate","hybrid-sleep","lock")
+    login_conf_d=os.getenv("TEST_LOGIN_CONF_D", default="/etc/systemd/logind.conf.d")
+    conf=os.path.join(login_conf_d, "00-snap-core.conf")
+
+    if not os.path.isdir(login_conf_d):
+        os.makedirs(login_conf_d)
+    if not action in valid_actions:
+        raise Exception("invalid action {!r} supplied for system.power-key-action option".format(action))
+    with atomic(conf) as fp:
+        fp.write("[Login]\nHandlePowerKey={}\n".format(action))
+
+def switch_service(service, enable):
+    srv = "{}.service".format(service)
+    if enable == "true":
+ 	# A simple `systemctl disable --now ssh.service` doesn't work
+ 	# and fails with an error message. Because of that we're going
+ 	# in two steps here and disable first and then stopping the
+ 	# service unit.
+        systemctl("disable", srv)
+        systemctl("stop", srv)
+    elif enable == "false":
+ 	# When the unit is already enabled but not active a call with
+ 	# --now does start the unit so we have to check for that  case
+ 	# and explicitly start the unit.
+        if systemctl("is-enabled", "--quiet", srv).returncode != 0:
+            systemctl("enable", srv, check=True)
+        if systemctl("is-active", "--quiet", srv).returncode != 0:
+            systemctl("start", srv, check=True)
+    else:
+        raise Exception("Invalid value {!r} provided for option {}".format(enable, srv))
+
+def update_pi_config(path, config):
+    to_write=[]
+    needs_write=False
+    with open(path) as r:
+        found=set()
+        rx=re.compile(r'^[ \t]*?(?P<is_comment>#?)[ \t#]*?(?P<key>{})=(?P<old_value>.*)$'.format("|".join(config)))
+        for line in r:
+            m=rx.match(line)
+            if m is not None:
+                key=m.group("key")
+                was_comment=m.group("is_comment") == "#"
+                old_value=m.group("old_value")
+                found.add(key)
+                if config[key]:
+                    if was_comment or old_value != config[key]:
+                        line="{}={}\n".format(key, config[key])
+                        needs_write=True
+                else:
+                    line="#{}={}\n".format(key, old_value)
+                    if not was_comment:
+                        needs_write=True
+            to_write.append(line)
+        for key in set(config)-found:
+            if config[key]:
+                needs_write=True
+                to_write.append("{}={}\n".format(key, config[key]))
+    if needs_write:
+        with atomic(path) as w:
+            for line in to_write:
+                w.write(line)
+
+if __name__ == "__main__":
+    if not core_support_available():
```

Comment:
You may want to wrap that in `main()` as otherwise each assignment below becomes a global that is visible in other functions.